### PR TITLE
user account not required

### PR DESCRIPTION
**Pull request by hannahkm**
[feature] : user can use the app without registering for an account

Summary: previously, the user would open the app and have to sign up for an account/log into an existing account. This PR changes this and lets them access the entire app without having to use an account.

Changes in this PR:
- Main activity changed to be MainActivity (Dashboard fragment)
- If the current user doesn't exist, a guest account is made for the current user
- User profile page shows a sign up button instead of a log out button when user is using a guest account
- Signing up for an account saves any itineraries, destinations, or favorites made while in the guest account

Callouts/Follow-ups:
- I think this PR also fixes a small visual glitch that occurred previously when the app started up; the user would see the login page before the dashboard if they are logged in

### CODE REVIEW

File: app/src/main/java/com/example/fbufinalapp/MainActivity.java [hunk 1]
```diff
@@ -38,6 +41,26 @@ protected void onCreate(Bundle savedInstanceState) {
         View view = binding.getRoot();
         setContentView(view);
 
+        if (ParseUser.getCurrentUser() == null){
```
Comment (Jibola) [inline app/src/main/java/com/example/fbufinalapp/MainActivity.java:44]: Need some clarification on how this part of the code works. 

It checks if no user is there. 

If no user is there, it creates a default "guest" user. 

1. Guest users cannot have their itineraries persist in state, correct? 
2. When/how often does this code run in the actual flow?

File: app/src/main/java/com/example/fbufinalapp/MainActivity.java [hunk 2]
```diff
@@ -38,6 +41,26 @@ protected void onCreate(Bundle savedInstanceState) {
...
+            ParseUser user = new ParseUser();
+            user.setUsername("Guest");
+            user.setPassword("default");
+            user.put("favorites", new ArrayList<>());
+            user.put("itineraries", new ArrayList<>());
```
Comment (Jibola) [inline app/src/main/java/com/example/fbufinalapp/MainActivity.java:49]: A good habit here would be encapsulating this in its own method called something like 
`createGuestUser` or something.

File: app/src/main/java/com/example/fbufinalapp/MainActivity.java [hunk 3]
```diff
@@ -38,6 +41,26 @@ protected void onCreate(Bundle savedInstanceState) {
...
+
+            user.signUpInBackground(e -> {
+                if (e == null) {
+                    Log.i("MainActivity", "signing up!");
+                    // Hooray! Let them use the app now.
```
Comment (Jibola) [inline app/src/main/java/com/example/fbufinalapp/MainActivity.java:54]: Both of these lines can be removed

File: app/src/main/java/com/example/fbufinalapp/MainActivity.java [hunk 4]
```diff
@@ -38,6 +41,26 @@ protected void onCreate(Bundle savedInstanceState) {
...
+                    Toast.makeText(MainActivity.this, "Welcome!", Toast.LENGTH_SHORT).show();
+                } else {
+                    // Sign up didn't succeed. Look at the ParseException
+                    // to figure out what went wrong
```
Comment (Jibola) [inline app/src/main/java/com/example/fbufinalapp/MainActivity.java:58]: These comments can be removed

File: app/src/main/java/com/example/fbufinalapp/ProfileFragment.java [hunk 5]
```diff
@@ -47,6 +49,9 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
         binding = FragmentProfileBinding.inflate(getLayoutInflater(), container, false);
         View view = binding.getRoot();
 
+        currentUser = ParseUser.getCurrentUser();
+        newUser = (currentUser.getEmail() == null);
```
Comment (Jibola) [inline app/src/main/java/com/example/fbufinalapp/ProfileFragment.java:53]: Looks like these two variables are unused?

File: app/src/main/java/com/example/fbufinalapp/SignUpActivity.java [hunk 6]
```diff
@@ -48,25 +54,40 @@ public void SignUp(){
...
-            }
-        });
+
+        if (newUser){
+            user.put("favorites", new ArrayList<>());
+            user.put("itineraries", new ArrayList<>());
+
+            user.signUpInBackground(e -> {
+                if (e == null) {
+                    // Hooray! Let them use the app now.
```
Comment (Jibola) [inline app/src/main/java/com/example/fbufinalapp/SignUpActivity.java:69]: comment not needed

File: app/src/main/java/com/example/fbufinalapp/SignUpActivity.java [hunk 7]
```diff
@@ -48,25 +54,40 @@ public void SignUp(){
...
+                    Intent i = new Intent(SignUpActivity.this, MainActivity.class);
+                    Toast.makeText(SignUpActivity.this, "Welcome!", Toast.LENGTH_SHORT).show();
+                    startActivity(i);
+                } else {
+                    // Sign up didn't succeed. Look at the ParseException
+                    // to figure out what went wrong
```
Comment (Jibola) [inline app/src/main/java/com/example/fbufinalapp/SignUpActivity.java:75]: Comment not needed

File: app/src/main/java/com/example/fbufinalapp/MainActivity.java (hunk 1 above)
Comment (hannahkm) [reply app/src/main/java/com/example/fbufinalapp/MainActivity.java:44]: Yes! So this line will check if there is a user current logged in. If there is no user, it'll create that guest account. 
1. At the moment, their itineraries do persist across states and when they sign up for a real account. Maybe this isn't how guest accounts usually work? 
2. The code runs once at the very beginning, before the rest of the app loads! I don't think it's run again until the app restarts

**Pull request merged**